src/bumper-core.coffee

```diff
@@ -16,7 +16,62 @@
 ) @, ->
 
   class BumperCore
-    version: '3.0.1'
+    # Maintain bumper version here...
+    version: '3.1.0'
+
+    # convert strings into a given javascript class type
+    #
+    castType: (string, type) ->
+      if type
+        switch type
+          when 'boolean'
+            switch string
+              when 'true' then true
+              when 'false' then false
+          when 'integer'
+            parseInt string
+          when 'float'
+            parseFloat string
+          else
+            string
+
+      # attempt detection if no type is passed
+      else
+        switch string
+          when 'true' then true
+          when 'false' then false
+          else string
+
+    # default global options object
+    Options: {}
+
+    # expose Module class to easily extend modules with
+    #
+    Module: class BumperModule
+      # default options object
+      options: {}
+
+      constructor: ->
+        # turn the class name into an array of class namespaces
+        # e.g. BumperFooBar => ['Foo', 'Bar']
+        names = @constructor
+          .name
```

Comment:
Got it, but that still is a 'no-two-name-key' rule (ie, you can never have Bumper->AwesomeModule->Image, it will always be Bumper->Awesome->Module->Image, which i guess is fine, but something to emphasize